dirty_cat/similarity_encoder.py

```diff
@@ -199,6 +244,10 @@ def __init__(self, similarity='ngram', ngram_range=(2, 4),
         if categories == 'auto' and n_prototypes is not None:
             warnings.warn('n_prototypes parameter ignored with category type \'auto\'')
 
+        self._vectorizers = None
+        self._vocabulary_count_matrices = None
+        self._vocabulary_ngram_counts = None
```

Comment:
These variable names should end with an underscore to indicate that they are derived from data (scikit-learn conventions)

Reply:
Also, in scikit-learn, we tend not to instantiate variables in init, but in fit